File: src/rustycage/editors/RustEditor.java
package rustycage.editors;

import org.eclipse.ui.editors.text.TextEditor;

public class RustEditor extends TextEditor {

    public RustEditor() {
		super();
		setSourceViewerConfiguration(new RustConfiguration());
		//setDocumentProvider(new RustDocumentProvider());
	}

	public void dispose() {
		super.dispose();
	}

    @Override
    protected void createActions() {
        super.createActions();
    }
}

File: rustycage-plugin/src/org/rustycage/editors/RustEditor.scala
package org.rustycage.editors

import org.eclipse.ui.editors.text.TextEditor
import org.eclipse.core.resources.{IResource, IFile}
import org.eclipse.core.runtime.IPath

class RustEditor extends TextEditor {

  setSourceViewerConfiguration(new RustConfiguration())
  
  override def dispose() {
    super.dispose()
  }
  
  override def createActions() {
    super.createActions()
  }

  override def setFocus() {
    super.setFocus()
  }

  def getFilePath() : IResource = {
    getEditorInput.getAdapter(classOf[IResource]).asInstanceOf[IResource]
  }

}
